lib/Sema/TypeCheckAccess.cpp

```diff
@@ -365,9 +368,16 @@ void AccessControlCheckerBase::checkGenericParamAccess(
   if (minAccessScope.isPublic())
     return;
 
+  // FIXME: Promote these to an error in the next -swift-version break.
+  if (isa<SubscriptDecl>(owner) || isa<TypeAliasDecl>(owner))
+    downgradeToWarning = DowngradeToWarning::Yes;
+
   if (checkUsableFromInline) {
-    auto diagID = diag::generic_param_usable_from_inline;
     if (!TC.Context.isSwiftVersionAtLeast(5))
+      downgradeToWarning = DowngradeToWarning::Yes;
+
+    auto diagID = diag::generic_param_usable_from_inline;
+    if (downgradeToWarning == DowngradeToWarning::Yes)
```

Comment:
I wouldn't do it in this PR even if you hadn't merged it already, but it might make sense to build something in to the `DiagnosticEngine` that downgrades errors to warnings if the language version is not high enough.

```cpp
TC.diagnoseLanguageChange(/*after:*/5, owner, diag::generic_param_usable_from_inline ...)
```

Reply:
Yeah, that's come up before. The main barrier in this case is that the text for these is slightly different (one is phrased as "oh, you shouldn't do this" and the other is "this doesn't work"), but maybe that could be abstracted out too.